### PR TITLE
Fix click sorting

### PR DESCRIPTION
**Pull request by metalgearsloth**
Reverse is redundant as that's what the sort is for. Fixes https://github.com/space-wizards/space-station-14/issues/11131

:cl:
- fix: Fix click sorting for entities with the same drawdepth.

### CODE REVIEW

File: Content.Client/Gameplay/GameplayStateBase.cs [hunk 1]
```diff
@@ -94,7 +93,7 @@ public ClickableEntityComparer(IEntityManager entities)
             public int Compare((EntityUid clicked, int depth, uint renderOrder) x,
                 (EntityUid clicked, int depth, uint renderOrder) y)
             {
-                var val = x.depth.CompareTo(y.depth);
+                var val = y.depth.CompareTo(x.depth);
```
Comment (ElectroJr) [inline Content.Client/Gameplay/GameplayStateBase.cs:96]: you also need to reverse all the other checks further down

File: Content.Client/Gameplay/GameplayStateBase.cs [hunk 2]
```diff
@@ -94,7 +93,7 @@ public ClickableEntityComparer(IEntityManager entities)
             public int Compare((EntityUid clicked, int depth, uint renderOrder) x,
                 (EntityUid clicked, int depth, uint renderOrder) y)
```
Comment (ElectroJr) [inline Content.Client/Gameplay/GameplayStateBase.cs:94]: Oher than being negated/reversed, this function should match the order in which sprites are drawn, so the top sprite is always preferred. E..g., the Y-sorting here should really be using the sprite bounding box y-sorting used in `SpriteDrawingOrderComparer` (bounding box calculated in `ProcessSpriteEntities()`)

**Comment by metalgearsloth** [comment]
Good for rebiew again

**Review by AJCM-git — APPROVED** [review]
I dont dare trying to understand `GameplayStateBase.cs` but the rest looks good

**Comment by ElectroJr** [comment]
Content Client_lI7AWOQitw
Ther are still some issues. E.g., in this image the click target should be the trash, but its targeting the bowl instead.

**Pull request merged**